server/routes/record-of-oral/index.ts

```diff
@@ -34,32 +34,33 @@ export default function Index(
 
   const get = (path: string, handler: RequestHandler) => router.get(routePrefix(path), asyncMiddleware(handler))
   const post = (path: string, handler: RequestHandler) => router.post(routePrefix(path), asyncMiddleware(handler))
-  const getAndPost = (path: string, handler: BaseController) => {
-    get(path, handler.get)
-    post(path, handler.post)
-  }
 
   router.get('/oralReport/:reportId/:section?', (req, res) => {
     const { reportId, section } = req.params
     res.redirect(301, `/record-of-oral/${reportId}${section ? `/${section}` : ''}`)
   })
 
   get('/:reportId', new LandingPageController(reportService, communityService, null, preSentenceToDeliusService).get)
-
-  getAndPost('/:reportId/offender-details', new OffenderDetailsController(reportService, communityService))
-  getAndPost('/:reportId/court-details', new CourtDetailsController(reportService, communityService))
-  getAndPost('/:reportId/offence-details', new OffenceDetailsController(reportService, communityService))
-  getAndPost('/:reportId/offence-analysis', new OffenceAnalysisController(reportService, communityService))
-  getAndPost('/:reportId/offender-assessment', new OffenderAssessmentController(reportService, communityService))
-  getAndPost('/:reportId/risk-assessment', new RiskAssessmentController(reportService, communityService))
-  getAndPost('/:reportId/proposal', new ProposalController(reportService, communityService))
-  getAndPost('/:reportId/sources-of-information', new SourcesOfInformationController(reportService, communityService))
-  getAndPost('/:reportId/sign-report', new SignReportController(reportService, communityService, eventService))
-
+  get('/:reportId/offender-details', new OffenderDetailsController(reportService, communityService).get)
```

Comment:
I believe it is still the same case except get handlers do not access post handlers' class variables but the same get function services all the user requests here which has access to all the shared variables and it uses them. If we go this way, we should be creating a hanlder function that creates a new object per user request like this https://github.com/ministryofjustice/pre-sentence-service/pull/186. The object serves the request and gets destroyed.

Reply:
I've refactored this now